### PR TITLE
Back-end delete feature

### PR DESCRIPTION
**Pull request by anthony23334**
(MOST CHANGED FILES ARE FROM THE GOOGLE FORMAT JOB)
- Added removeMessage for messageStore to handle the HttpRequest from the chatServlet. 
- Also, this function calls the PersistentStorageAgent in order to update the database by removing the message. 
- Added Tests. 
- THE NEXT PR WILL ADDRESS ACTIVITYFEED UPDATE

### CODE REVIEW

File: src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java [hunk 1]
```diff
@@ -148,6 +148,11 @@ public void testSaveAndLoadMessages() throws PersistentDataStoreException {
     Assert.assertEquals(authorTwo, resultMessageTwo.getAuthorId());
     Assert.assertEquals(contentTwo, resultMessageTwo.getContent());
     Assert.assertEquals(creationTwo, resultMessageTwo.getCreationTime());
+
+    // confirm that we deleted all the messages
+    persistentDataStore.deleteFrom(inputMessageOne);
+    persistentDataStore.deleteFrom(inputMessageTwo);
+    Assert.assertTrue(persistentDataStore.loadActivities().isEmpty());
```
Comment (donnelldebnam) [inline src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java]: Nice test here.

File: src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java [hunk 2]
```diff
@@ -109,4 +111,13 @@ public void testWriteThroughActivity() {
...
   }
+
+  @Test
+  public void testDeleteFromMessage() {
+    Message message =
+        new Message(
+            UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(), "test content", Instant.now());
+    persistentStorageAgent.deleteFrom(message);
+    Mockito.verify(mockPersistentDataStore).deleteFrom(message);
+  }
```
Comment (donnelldebnam) [inline src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java]: What exactly is this testing?

File: src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java (hunk 1 above)
Comment (anthony23334) [reply src/test/java/codeu/model/store/persistence/PersistentDataStoreTest.java]: Thanks

File: src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java (hunk 2 above)
Comment (anthony23334) [reply src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java]: I was a bit confused too, but I tried to follow the existing format. I believe that this code verifies that by calling `persistentStorageAgent.deleteFrom(message);` the datastore's delete function is also called. Basically, it ensures there is a clear connection between the agent and the datastore. I think.

Comment (donnelldebnam) [reply src/test/java/codeu/model/store/persistence/PersistentStorageAgentTest.java]: Got it.

**Pull request merged**